packages/ckeditor5-image/src/imagestyle/imagestylecommand.js

```diff
@@ -56,15 +58,21 @@ export default class ImageStyleCommand extends Command {
 	refresh() {
 		const element = this.editor.model.document.selection.getSelectedElement();
 
-		this.isEnabled = isImage( element );
+		this.isEnabled = isImage( element ) || isImageInline( element );
 
 		if ( !element ) {
 			this.value = false;
 		} else if ( element.hasAttribute( 'imageStyle' ) ) {
 			const attributeValue = element.getAttribute( 'imageStyle' );
-			this.value = this.styles[ attributeValue ] ? attributeValue : false;
+			const arrangement = this._arrangements.get( attributeValue );
+
+			if ( arrangement && ( !arrangement.modelElement || arrangement.modelElement === element.name ) ) {
```

Comment:
With changes to `viewToModelStyleAttribute()` there will be no way to load an incorrect arrangement into the editor, so this condition is bogus.

Reply:
Done.